test/gpu/amdgpu.jl

```diff
@@ -6,10 +6,18 @@ using LinearOperators, AMDGPU
   B = ROCArray(rand(Float32, 10, 10))
   C = ROCArray(rand(Float32, 20, 20))
   M = BlockDiagonalOperator(A, B, C)
+  v = ROCArray(rand(5))
+
 
   v = ROCArray(rand(Float32, 35))
   y = M * v
   @test y isa ROCArray{Float32}
 
+  @test LinearOperators.storage_type(A) == LinearOperators.storage_type(adjoint(A))
+  @test LinearOperators.storage_type(A) == LinearOperators.storage_type(transpose(A))
+  @test LinearOperators.storage_type(A) == LinearOperators.storage_type(adjoint(A))
+  @test LinearOperators.storage_type(Diagonal(v)) == typeof(v)
```

Comment:
or alternately, move these to before the assignment `v = ROCArray(rand(Float32, 35))`